internal/plugin/reflike.go

```diff
@@ -0,0 +1,45 @@
+package plugin
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"go.jetpack.io/devbox/nix/flake"
+)
+
+// RefLike is like a flake ref, but in some ways more general. It can be used
+// to reference other types of files, e.g. devbox.json.
```

Comment:
is the only difference that flake ref refer to directories, whereas here it can refer to a specific file? If so, could we call that out more strongly?

Reply:
flake refs are specific to nix flakes and I wanted to avoid modifying that. The only real difference here is that there's an optional default filename the reflike referres to. In the case of plugins is `devbox.json` but I want to leave the door open to using reflikes for composing multiple projects as well if that is something we want to support in the future (In that case the filename is probably a `devbox.json`)